cmd/migration-tool/main.go

```diff
@@ -0,0 +1,524 @@
+			if err := opt.Run(); err != nil {
+				klog.Exitf("Run error: %v", err)
+			}
+		},
+	}
+
+	flagset := cmd.Flags()
+	flagset.BoolVar(&opt.Execute, "execute", opt.Execute, "Create ReleasePayload resources on currently configured cluster")
+	flagset.StringVar(&opt.Namespace, "namespace", opt.Namespace, "Namespace where the image stream, containing releases, resides")
+	flagset.StringVar(&opt.Name, "name", opt.Name, "Name of the image stream, containing releases, to process")
+	flagset.StringVar(&opt.ProwNamespace, "prow-namespace", opt.ProwNamespace, "The namespace where Prow jobs are executed.")
+	flagset.StringVar(&opt.Location, "location", opt.Location, "The path to a folder containing the release payloads from a previous invocation.")
+	flagset.AddGoFlag(original.Lookup("v"))
+
+	if err := cmd.Execute(); err != nil {
+		klog.Exitf("Execute error: %v", err)
+	}
+}
+
+func (o *options) Run() error {
+	ctx := context.Background()
+	ctxWithCancel, cancelFunction := context.WithCancel(ctx)
+
+	defer func() {
+		cancelFunction()
+	}()
+
+	stopCh := make(chan bool)
+
+	inClusterCfg, err := loadClusterConfig(&clientcmd.ConfigOverrides{})
+	imageClient, err := imageclientset.NewForConfig(inClusterCfg)
+	if err != nil {
+		return fmt.Errorf("unable to create image client: %v", err)
+	}
+
+	releasePayloadClient, err := versioned.NewForConfig(inClusterCfg)
+	if err != nil {
+		return fmt.Errorf("unable to create release payload client: %v", err)
+	}
+
+	go mainProcessLoop(ctxWithCancel, stopCh, imageClient.ImageV1(), releasePayloadClient.ReleaseV1alpha1(), o.Namespace, o.Name, o.ProwNamespace, o.Execute, o.Location)
+
+	select {
+	case <-ctx.Done():
+		klog.V(5).Infof("Context has been cancelled")
+	case <-stopCh:
+		klog.V(5).Infof("mainProcessLoop returned")
+	}
+	return nil
+}
+
+func loadClusterConfig(overrides *clientcmd.ConfigOverrides) (*rest.Config, error) {
+	cfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(clientcmd.NewDefaultClientConfigLoadingRules(), overrides)
+	clusterConfig, err := cfg.ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load client configuration: %v", err)
+	}
+	return clusterConfig, nil
+}
+
+func mainProcessLoop(ctx context.Context, stopCh chan bool, imageClient imageclient.ImageV1Interface, releasePayloadClient releasev1alpha1.ReleaseV1alpha1Interface, namespace, name, prowNamespace string, execute bool, location string) {
+	ctxWithTimeout, cancelFunction := context.WithTimeout(ctx, time.Duration(15)*time.Second)
+
+	defer func() {
+		cancelFunction()
+		stopCh <- true
+	}()
+
+	var errors []error
+	var payloads []*v1alpha1.ReleasePayload
+	controller := newController(ctxWithTimeout, imageClient, namespace, name, prowNamespace)
+
+	if !execute {
+		klog.Infof("Running in Dry-Run mode.  No objects will be created on cluster.")
+	}
+
+	if len(location) > 0 {
+		payloads = readPayloads(location)
+
+		if execute {
+			errors = createPayloads(releasePayloadClient, payloads)
```

Comment:
I don't think you want this command to do this.  I think you want only the controller to do the write or you'll have a hard time motivating the transition to a controller from a command.